sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -2072,6 +2074,350 @@ private Response<RecognizeLinkedEntitiesResultCollection> RecognizeLinkedEntitie
+                throw;
+            }
+
+        }
+
+        /// <summary> Cancel healthcare prediction job. </summary>
+        /// <param name="operation"> Job ID. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        #pragma warning disable AZC0015 // Unexpected client method return type.
+        public virtual string StartCancelHealthJob(HealthcareOperation operation, CancellationToken cancellationToken = default)
+        #pragma warning restore AZC0015 // Unexpected client method return type.
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TextAnalyticsClient)}.{nameof(StartCancelHealthJobAsync)}");
+            scope.Start();
+
+            try
+            {
+                ResponseWithHeaders<TextAnalyticsCancelHealthJobHeaders> response = _serviceRestClient.CancelHealthJob(new Guid(operation.Id), cancellationToken);
+
+                return response.Headers.OperationLocation;
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets collection of healthcare entities from the HealthOperation using async pageable.
+        /// </summary>
+        /// <param name="operation"></param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>A collection of <see cref="RecognizeHealthcareEntitiesResultCollection"/> items.</returns>
+        public virtual AsyncPageable<DocumentHealthcareResult> GetHealthcareEntities(HealthcareOperation operation, CancellationToken cancellationToken = default)
+        {
+            async Task<Page<DocumentHealthcareResult>> FirstPageFunc(int? pageSizeHint)
+            {
+                using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TextAnalyticsClient)}.{nameof(GetHealthcareEntities)}");
+                scope.Start();
+
+                try
+                {
+                    Response<RecognizeHealthcareEntitiesResultCollection> response = await operation.WaitForCompletionAsync().ConfigureAwait(false);
+
+                    RecognizeHealthcareEntitiesResultCollection result = operation.Value;
+                    return Page.FromValues(result.AsEnumerable(), operation.NextLink, response.GetRawResponse());
+                }
+                catch (Exception e)
+                {
+                    scope.Failed(e);
+                    throw;
+                }
+            }
+            async Task<Page<DocumentHealthcareResult>> NextPageFunc(string nextLink, int? pageSizeHint)
+            {
+                using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TextAnalyticsClient)}.{nameof(GetHealthcareEntities)}");
+                scope.Start();
+
+                try
+                {
+                    int top = 1;
```

Comment:
this is a lot of magic :( 
Could you add comments on how the string that you are parsing looks like?
also, any reason why you are not using regex like we do, for example, when resolving the relations object? it is more clean

Reply:
I wasnt able to get regex to work for the URL. Then I moved to this one. WIll add comments. If you have something for reference I can give it a try

Reply:
we can leave it with split but then it has to have validations to make sure we are not going to get a NullReferenceException

Reply:
Could u create an issue to look more into the parsing for this pagination in the future?